src/back/TicketManager/src/PaymentService/TicketManager.PaymentService.Services/ApiKeyAuth/ApiKeyAuthorization.cs

```diff
@@ -0,0 +1,28 @@
+using FastEndpoints;
+using FluentValidation.Results;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using TicketManager.PaymentService.Services.Configuration;
+
+namespace TicketManager.PaymentService.Services.ApiKeyAuth;
+
+public class ApiKeyAuthorization<TRequest> : IPreProcessor<TRequest>
+{
+    private const string ApiKeyHeaderName = "pay_api_key";
+    private PaymentServiceConfiguration configuration; 
+    
+    public ApiKeyAuthorization(PaymentServiceConfiguration configuration) {
+        this.configuration = configuration;
+    }
+    public Task PreProcessAsync(TRequest req, HttpContext context, List<ValidationFailure> failures, CancellationToken ct)
+    {
+        var headerPresent = context.Request.Headers.TryGetValue(ApiKeyHeaderName, out var extractedApiKey);
+        var apiKey = configuration.ApiKey;
+        
+        if (!headerPresent || !apiKey.Equals(extractedApiKey)) {
+            return context.Response.SendForbiddenAsync(ct);
```

Comment:
You're still sending 403 instead of 401. But because it's not specified in documentation, it doesn't really matter.

Reply:
Ah I thought that the commend above meant to change it from 401 to 403.

> Here you return status code 403, not 401.

Reply:
Addressed in 3caa1993.